sklearn/ensemble/stacking.py

```diff
@@ -0,0 +1,265 @@
+
+        Note that this behavior is different from ``fit().transform()`` as it
+        will return the cross validation predictions instead. Read more in the
+        :ref:`User Guide <stacking_transformer>`.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape=(n_samples, n_features)
+            Input data used to build forests. Use ``dtype=np.float32`` for
+            maximum efficiency.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        **fit_params : parameters to be passed to the base estimator.
+
+        Returns
+        -------
+        X_transformed : sparse matrix, shape=(n_samples, n_out)
+            Transformed dataset.
+        """
+        self.estimator_ = clone(self.estimator).fit(X, y, **fit_params)
+
+        preds = cross_val_predict(clone(self.estimator), X, y, cv=self.cv,
+                                  method=self._method_name(),
+                                  n_jobs=self.n_jobs, fit_params=fit_params)
+
+        if preds.ndim == 1:
+            preds = preds.reshape(-1, 1)
+
+        return preds
+
+
+def _identity(x):
+    return x
+
+
+def _identity_transformer():
+    """Contructs a transformer that returns its input unchanged"""
+    return FunctionTransformer(_identity, accept_sparse=True)
+
+
+def make_stack_layer(estimators, restack=False, cv=3, method='auto',
+                     n_jobs=1, n_cv_jobs=1, transformer_weights=None):
+    """ Construct a single layer for model stacking
+
+    Read more in the :ref:`User Guide <stacking>`.
+
+    Parameters
+    ----------
+    estimators : list
+        List of (name, predictor) tuples to be used in stacking.
+
+    restack : bool, optional (default=False)
+        Whether input should be concatenated to the transformation.
+
+    cv : int, cross-validation generator or an iterable, optional (default=3)
+        Determines the cross-validation splitting strategy to be used for
+        generating features to train the next layer on the stacked ensemble or,
+        more specifically, during ``fit_transform``.
+
+        Possible inputs for cv are:
+
+        - None, to use the default 3-fold cross-validation,
+        - integer, to specify the number of folds.
+        - An object to be used as a cross-validation generator.
+        - An iterable yielding train/test splits.
+
+        For integer/None inputs, if the estimator is a classifier and ``y`` is
+        either binary or multiclass,
+        :class:`sklearn.model_selection.StratifiedKFold` is used. In all
+        other cases, :class:`sklearn.model_selection.KFold` is used.
+
+        Refer :ref:`User Guide <cross_validation>` for the various
+        cross-validation strategies that can be used here.
+
+    method : string, optional (default='auto')
+        Invokes the passed method name of the estimators. If the method is
+        ``auto``, will try to invoke ``predict_proba`` or ``predict`` in that
```

Comment:
you forgot `decision_function`.
Please use consistent docstrings between `StackingTransformer` and `make_stack_layer`.